From 3fdb2265820f0d3570e844069f5d1113ac1796fe Mon Sep 17 00:00:00 2001
From: Jon West <electrikjesus@gmail.com>
Date: Fri, 17 Aug 2018 23:51:45 -0400
Subject: [PATCH] Disable AVX support in bcc

Orig From: Yang Ni <yangni@google.com>
Date: Mon, 13 Jun 2016 11:18:40 -0700
Subject: [PATCH] Disable AVX support in bcc

Bug: 28879581

LLVM generates AVX code for long3 load/store, which treats a long3 value
as 256 bits, while the RS runtime only allocates 192 bits for a long3
element in an allocation.

This CL disables AVX codegen to avoid this issue.

Change-Id: I61875754c608fe2e91a3c2b1619d652e49c5b388
---
 lib/CompilerConfig.cpp | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/lib/CompilerConfig.cpp b/lib/CompilerConfig.cpp
index 3ba8498..d35c998 100644
--- a/lib/CompilerConfig.cpp
+++ b/lib/CompilerConfig.cpp
@@ -202,6 +202,11 @@ bool CompilerConfig::initializeArch() {
     // generic fallback for 32bit x86 targets
     setCPU("atom");
 #endif // DEFAULT_X86_CODEGEN && !__HOST__
+
+	// LLVM generates AVX code that treats a long3 as 256 bits, while
+    // RenderScript considers a long3 192 bits (http://b/28879581)
+    attributes.push_back("-avx");
+    
     break;
 #endif  // PROVIDE_X86_CODEGEN
 
@@ -222,6 +227,11 @@ bool CompilerConfig::initializeArch() {
       setCodeModel(llvm::CodeModel::Medium);
     }
     getTargetOptions().UseInitArray = true;
+    
+    // LLVM generates AVX code that treats a long3 as 256 bits, while
+    // RenderScript considers a long3 192 bits (http://b/28879581)
+    attributes.push_back("-avx");
+    
     break;
 #endif  // PROVIDE_X86_CODEGEN
 
-- 
2.17.1

